src/utils.ts

```diff
@@ -158,4 +159,67 @@ export function createDidPayload(verificationMethods: VerificationMethod[], veri
             authentication: verificationKeys.map(key => key.keyId)
         }
     )
+}
+
+export function createDidPayloadWithSignInputs(seed?: string, keys?: IKeyPair[]) {
+    if(seed && keys) throw new Error ('Only one of seed or keys should be passed as an argument')
+    
+    if(!keys) {
+        keys = [seed ? createKeyPairBase64(seed) : createKeyPairBase64()]
+    }
+
+    const verificationMethodTypes = keys.map((key) => !key.algo || key.algo == MethodSpecificIdAlgo.Base58 ? VerificationMethods.Base58 : VerificationMethods.JWK)
+    const verificationKeys = keys.map((key, i) => createVerificationKeys(key, key.algo || MethodSpecificIdAlgo.Base58, `key-${i}`))
+    const verificationMethod = createDidVerificationMethod(verificationMethodTypes, verificationKeys)
+    
+    let payload : Partial<MsgCreateDidPayload> = {
+        id: verificationKeys[0].didUrl,
+        controller: verificationKeys.map(key => key.didUrl),
+        verificationMethod: verificationMethod,
+        authentication: verificationKeys.map(key => key.keyId),
+    }
+
+    const keyHexs = keys.map((key)=>convertKeyPairtoTImportableEd25519Key(key))
+    const signInputs = keyHexs.map((key)=>createSignInputsFromImportableEd25519Key(key, verificationMethod))
+
+    return { didPayload: MsgCreateDidPayload.fromPartial(payload), keys, signInputs }
+}
+
+export function convertKeyPairtoTImportableEd25519Key(keyPair: IKeyPair) : TImportableEd25519Key {
+    return {
+        type: 'Ed25519',
+        privateKeyHex: toString(fromString(keyPair.privateKey, 'base64'), 'hex'),
+        kid: 'kid',
+        publicKeyHex: toString(fromString(keyPair.publicKey, 'base64'), 'hex')
+    }
+}
+
+export function createSignInputsFromKeyPair(didDocument: IdentifierPayload, keys: IKeyPair[]) {
+    const keyHexs = keys.map((key)=>convertKeyPairtoTImportableEd25519Key(key))
+    const signInputs = keyHexs.map((key)=>createSignInputsFromImportableEd25519Key(key, didDocument.verificationMethod!))
+    return signInputs
+}
+
+export enum DidDocumentOperation {
+    Set = 'setDidDocument',
+    Add = 'addToDidDocument',
+    Remove = 'removeFromDidDocument'
+}
+
+export function jsonConcat(o1: any, o2:any) {
+    for (var key in o2) {
+    if(Array.isArray(o1[key])) {
+        o1[key].push(...o2[key])
+    } else {
+        o1[key] = o2[key]
+    }}
+    return o1
+}
+
+export function jsonSubtract(o1: any, o2: any) {
+    for (var key in o2) {
+        if(o2[key] == o1[key]) {
+            delete(o1[key])
+        }
+    }
```

Comment:
Are those tools needed to be exported by our SDK?
I'd suggest to keep them defined in your application scope, since usage is limited to external and app-specific purposes.